### PR TITLE
OSDOCS#7674:LSO option to wipe LVs

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.15+

Issue: https://issues.redhat.com/browse/OSDOCS-7674

Link to docs preview: https://68130--docspreview.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-local#local-volume-cr_persistent-storage-local

QE review:
- [x] QE has approved this change.

Additional information:

**PTAL**:  @mpatlasov, @wsun1, @gcharot

### CODE REVIEW

File: modules/persistent-storage-local-create-cr.adoc [hunk 1]
```diff
@@ -46,23 +46,25 @@ spec:
...
 <3> The name of the storage class to use when creating persistent volume objects. The Local Storage Operator automatically creates the storage class if it does not exist. Be sure to use a storage class that uniquely identifies this set of local volumes.
-<4> The volume mode, either `Filesystem` or `Block`, that defines the type of local volumes.
+<4> Instructs Local Storage Operator on whether or not to erase the disk before trying to provision it. The default is "false" (disk is not erased prior to provisioning). Setting `forceWipeDevicesAndDestroyAllData` to "true" can be useful in scenarios where previous data can remain on disks that need to be re-used. In these scenarios, setting this field to true eliminates the need for administrators to erase the disks manually. Such cases can include single-node OpenShift (SNO) cluster environments where a node can be redeployed multiple time or when using OpenShift Container Storage (OCS), where previous data can remain on the disks planned to be consumed as object storage devices (OSDs).
```
Comment (skrthomas) [inline modules/persistent-storage-local-create-cr.adoc:58]:
```suggestion
<4> This setting instructs Local Storage Operator on whether or not to erase the disk before trying to provision it. The default is "false" (the disk is not erased prior to provisioning). Setting `forceWipeDevicesAndDestroyAllData` to "true" can be useful in scenarios where previous data can remain on disks that need to be re-used. In these scenarios, setting this field to true eliminates the need for administrators to erase the disks manually. Such cases can include single-node OpenShift (SNO) cluster environments where a node can be redeployed multiple times or when using OpenShift Container Storage (OCS), where previous data can remain on the disks planned to be consumed as object storage devices (OSDs).
```

Comment (skrthomas) [reply modules/persistent-storage-local-create-cr.adoc:58]: Same comment in the duplicated call out later in this PR

**Review by skrthomas — COMMENTED** [review]
A couple small comments, lgtm otherwise :) Nice work!

File: modules/persistent-storage-local-create-cr.adoc [hunk 2]
```diff
@@ -46,23 +46,25 @@ spec:
...
 <3> The name of the storage class to use when creating persistent volume objects. The Local Storage Operator automatically creates the storage class if it does not exist. Be sure to use a storage class that uniquely identifies this set of local volumes.
-<4> The volume mode, either `Filesystem` or `Block`, that defines the type of local volumes.
+<4> This setting instructs Local Storage Operator on whether or not to erase the disk before trying to provision it. The default is "false" (the disk is not erased prior to provisioning). Setting `forceWipeDevicesAndDestroyAllData` to "true" can be useful in scenarios where previous data can remain on disks that need to be re-used. In these scenarios, setting this field to true eliminates the need for administrators to erase the disks manually. Such cases can include single-node OpenShift (SNO) cluster environments where a node can be redeployed multiple times or when using OpenShift Container Storage (OCS), where previous data can remain on the disks planned to be consumed as object storage devices (OSDs).
```
Comment (gcharot) [inline modules/persistent-storage-local-create-cr.adoc:58]: we need to be more precise here. The wipe option calls wipefs which deletes the partition tables, it does not erase data per se. Technically data remains on the disks but it is ready to be "overwritten". For security transparency this should be explicitly mentioned. 

Also OCS does no exist anymore, it's called ODF now; openshift data foundation

cc @mpatlasov

File: modules/persistent-storage-local-create-cr.adoc [hunk 3]
```diff
@@ -90,16 +92,18 @@ spec:
...
 <3> The name of the storage class to use when creating persistent volume objects.
-<4> The volume mode, either `Filesystem` or `Block`, that defines the type of local volumes.
-<5> The path containing a list of local storage devices to choose from.
-<6> Replace this value with your actual local disks filepath to the `LocalVolume` resource `by-id`, such as `dev/disk/by-id/wwn`. PVs are created for these local disks when the provisioner is deployed successfully.
+<4> This setting instructs Local Storage Operator on whether or not to erase the disk before trying to provision it. The default is "false" (the disk is not erased prior to provisioning). Setting `forceWipeDevicesAndDestroyAllData` to "true" can be useful in scenarios where previous data can remain on disks that need to be re-used. In these scenarios, setting this field to true eliminates the need for administrators to erase the disks manually. Such cases can include single-node OpenShift (SNO) cluster environments where a node can be redeployed multiple times or when using OpenShift Container Storage (OCS), where previous data can remain on the disks planned to be consumed as object storage devices (OSDs).
```
Comment (gcharot) [inline modules/persistent-storage-local-create-cr.adoc:103]: same here
cc @mpatlasov

File: modules/persistent-storage-local-create-cr.adoc (hunk 2 above)
Comment (mpatlasov) [reply modules/persistent-storage-local-create-cr.adoc:58]: Agreed, we could borrow some language from wipefs man-page:

> wipefs can erase filesystem, raid or partition-table signatures (magic strings) from the specified device to make the signatures invisible for libblkid. wipefs does not erase the filesystem itself nor any other data from the device.

Three key points to highlight: 1) signatures (magic strings) are removed; 2) this makes disk ready to use for LSO provisioning; 3) no other data (other than signatures) erased.

Comment (lpettyjo) [reply modules/persistent-storage-local-create-cr.adoc:58]: Fixed. Please reivew.

File: modules/persistent-storage-local-create-cr.adoc (hunk 3 above)
Comment (lpettyjo) [reply modules/persistent-storage-local-create-cr.adoc:103]: Fixed. Please review.

**Comment by gcharot** [comment]
LGTM

**Comment by duanwei33** [comment]
@chao007  Could you help take a look?

**Pull request merged**